### PR TITLE
Add s3am

### PR DESCRIPTION
**Pull request by cket**
Continued from #80

### CODE REVIEW

File: s3am/test.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import subprocess
+import tempfile
+import unittest
+
+
+class TestS3AM(unittest.TestCase):
+
+    def test_docker_call(self):
+        out, err = check_docker_output(tool='quay.io/ucsc_cgl/s3am:1.0a1')
+        self.assertTrue('usage: s3am [--help] {upload,cancel,verify} ...')
```
Comment (hannes-ucsc) [inline s3am/test.py]: This is always True. You want `in out`

File: s3am/wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# activate s3am virtualenv
+source /opt/s3am/bin/activate
```
Comment (hannes-ucsc) [inline s3am/wrapper.sh]: Just invoke s3am by direct path. The venv will be activated on-the-fly.

File: s3am/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# install necessary dependencies
+RUN apt-get update && \
+    apt-get install -y python-dev \
+            gcc \
+            make \
+            libcurl4-openssl-dev \
+            python-virtualenv
+
+# create virtualenv and install s3am
+RUN virtualenv /opt/s3am && . /opt/s3am/bin/activate
```
Comment (hannes-ucsc) [inline s3am/Dockerfile]: Remove the sourcing of activate

File: s3am/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+RUN pip install s3am==2.0
```
Comment (hannes-ucsc) [inline s3am/Dockerfile]: Invoke pip as /opt/s3am/bin/pip and the venv will be activated on the fly.

File: s3am/test.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+def check_docker_output(tool):
```
Comment (hannes-ucsc) [inline s3am/test.py]: What does this give us over subprocess.check_output?

File: s3am/test.py (hunk 1 above)
Comment (hannes-ucsc) [reply s3am/test.py]: or `in err`

File: s3am/test.py (hunk 5 above)
Comment (hannes-ucsc) [reply s3am/test.py]: Other than not throwing CalledProcessException

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
@cket

**Comment by cket** [comment]
@hannes-ucsc - ready for review again

File: s3am/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# install necessary dependencies
+RUN apt-get update && \
+    apt-get install -y python-dev \
+            gcc \
+            make \
+            libcurl4-openssl-dev \
+            python-virtualenv
+
+# create virtualenv and install s3am
+RUN virtualenv /opt/s3am
```
Comment (hannes-ucsc) [inline s3am/Dockerfile]: Pass --no-download to virtualenv

File: s3am/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+RUN /opt/s3am/bin/pip install s3am==2.0
+
+# add wrapper script
+WORKDIR /opt/
+ADD wrapper.sh /opt/wrapper.sh
+
+ENTRYPOINT ["bash", "+x", "/opt/wrapper.sh"]
```
Comment (hannes-ucsc) [inline s3am/Dockerfile]: Lets omit the wrapper and make s3am the entry point. The perm fix is taken care of by docker_call now.

File: s3am/test.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import unittest
+
+
+class TestS3AM(unittest.TestCase):
+
+    def test_docker_call(self):
+        try:
+            out = check_output(['docker', 'run','quay.io/ucsc_cgl/s3am'], stderr=STDOUT)
+        except CalledProcessError as e:
+            print vars(e)
```
Comment (hannes-ucsc) [inline s3am/test.py]: ?

File: s3am/test.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python2.7
+# @author Frank Austin Nothaft fnothaft@berkeley.edu
```
Comment (hannes-ucsc) [inline s3am/test.py]: Remove this line

File: s3am/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:12.04
+
+MAINTAINER Frank Austin Nothaft
+MAINTAINER CJ Ketchum
```
Comment (hannes-ucsc) [inline s3am/Dockerfile]: Remove these two lines

File: s3am/Makefile [hunk 11]
```diff
@@ -0,0 +1,27 @@
+# Definitions
+build_tool = runtime-container.DONE
+name = quay.io/ucsc_cgl/s3am
+git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../s3am | cut -f1 -d " ")
+tag = 2.0--${git_commit}
+
+# Steps
```
Comment (hannes-ucsc) [inline s3am/Makefile]: Remove

File: s3am/Makefile [hunk 12]
```diff
@@ -0,0 +1,27 @@
+# Definitions
```
Comment (hannes-ucsc) [inline s3am/Makefile]: Remove

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Main requested change is removal of wrapper

**Comment by cket** [comment]
@hannes-ucsc ready

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**